identity-hub-client/src/test/java/IdentityHubClientImplTest.java

```diff
@@ -0,0 +1,111 @@
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Test
+    void getVerifiableCredentials() throws IOException {
+        VerifiableCredential credential = VerifiableCredential.Builder.newInstance().id(VERIFIABLE_CREDENTIAL_ID).build();
+
+        Interceptor interceptor = chain -> {
+            Request request = chain.request();
+            var replies = MessageResponseObject.Builder.newInstance().messageId(MESSAGE_ID_VALUE)
+                    .status(MessageStatus.OK).entries(List.of(credential)).build();
+            var responseObject = ResponseObject.Builder.newInstance()
+                    .requestId(FAKER.internet().uuid())
+                    .status(RequestStatus.OK)
+                    .replies(List.of(replies))
+                    .build();
+            var body = ResponseBody.create(OBJECT_MAPPER.writeValueAsString(responseObject), MediaType.get("application/json"));
+
+            Response response = new Response.Builder()
+                    .body(body)
+                    .request(request)
+                    .protocol(Protocol.HTTP_2)
+                    .code(200)
+                    .message("")
+                    .build();
+            return response;
+        };
+
+        var client = createClient(interceptor);
+        var credentials = client.getVerifiableCredentials(HUB_URL);
+        assertThat(credentials).usingRecursiveFieldByFieldElementComparator().containsExactly(credential);
+    }
+
+    @Test
+    void getVerifiableCredentialsServerError() {
+
+        Interceptor interceptor = chain -> {
+            Request request = chain.request();
+            var body = ResponseBody.create("{}", MediaType.get("application/json"));
+            Response response = new Response.Builder()
+                    .body(body)
+                    .request(request)
+                    .protocol(Protocol.HTTP_2)
+                    .code(500)
+                    .message("")
+                    .build();
+            return response;
+        };
+
+        var client = createClient(interceptor);
+
+        assertThatThrownBy(() -> client.getVerifiableCredentials(HUB_URL)).isInstanceOf(ServerException.class);
+    }
+
+    @Test
+    void getVerifiableCredentialsDeserializationError() {
+        Interceptor interceptor = chain -> {
+            Request request = chain.request();
+            var body = ResponseBody.create("{}", MediaType.get("application/json"));
+
+            Response response = new Response.Builder()
+                    .body(body)
+                    .request(request)
+                    .protocol(Protocol.HTTP_2)
+                    .code(200)
+                    .message("{}")
+                    .build();
+            return response;
+        };
+
+        var client = createClient(interceptor);
+        assertThatThrownBy(() -> client.getVerifiableCredentials(HUB_URL)).isInstanceOf(ValueInstantiationException.class);
```

Comment:
maybe generalise the assertion to `DatabindException`. That exception should then be documented in JavaDoc